pkg/scope/cluster.go

```diff
@@ -150,9 +152,25 @@ func (s *ClusterScope) setupProxmoxClient(ctx context.Context) (capmox.Client, e
 	tokenSecret := string(secret.Data["secret"])
 	url := string(secret.Data["url"])
 
-	// TODO, check if we need to delete tls config
+	tlsInsecure := string(secret.Data["insecure"]) != "false"
+	tlsRootCA := secret.Data["root_ca"]
+
+	rootCerts, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("loading system cert pool: %w", err)
+	}
+
+	if len(tlsRootCA) > 0 {
```

Comment:
A question which I do not know the answer to:

Hypothetically, if secret contains a chain, we can now no longer load it. I don't know if cluster-api allows for this, or if that's sane, but it's a question I have.

Reply:
Why can't we load it? 🤔

Reply:
This is just a hypothetical, but:
1. System store is empty, therefore `len(tlsRootCa) <= 0`
2. The certificate here is actually a self contained chain (not sane, not sure if possible).
Consequence: that no longer loads.

Reply:
`len(tlsRootCa)` is the len of the byte slice from the secret (so looking up whether there is something in the secret)